### PR TITLE
feat(core): support share edgeless mode

### LINKED ISSUE #3287: Support sharing doc as edgeless mode
After supporting public sharing, users can choose to publicly share the content in Page or Edgeless.

### PR DESCRIPTION
**Pull request by JimmFly**
Close #3287 

### Summary
📄🚀🔗

This pull request adds a feature to the frontend component of AFFiNE that allows the user to share a page in either `page` or `edgeless` mode, which affects the appearance and functionality of the page. It also adds the necessary GraphQL operations, types, and schema to support this feature in the backend, and updates the tests and the storybook stories accordingly.

*  Modify the `useIsSharedPage` hook to accept an optional `shareMode` argument and use the `getWorkspacePublicPagesQuery`, `publishPageMutation`, and `revokePublicPageMutation` from `@affine/graphql`

### CODE REVIEW

File: packages/frontend/core/src/hooks/affine/use-is-shared-page.ts [hunk 1]
```diff
@@ -1,57 +1,142 @@
...
+  getWorkspacePublicPagesQuery,
+  publishPageMutation,
+  revokePublicPageMutation,
 } from '@affine/graphql';
+import { useAFFiNEI18N } from '@affine/i18n/hooks';
 import { useMutation, useQuery } from '@affine/workspace/affine/gql';
+import { useSetAtom } from 'jotai';
 import { useCallback, useMemo } from 'react';
 
+export enum PublicPageMode {
```
Comment (forehalo) [inline packages/frontend/core/src/hooks/affine/use-is-shared-page.ts:13]: better use the one from 'affine/gql'

File: packages/frontend/component/src/components/share-menu/share-menu.tsx [hunk 2]
```diff
@@ -23,10 +25,18 @@ export interface ShareMenuProps<
...
+  currentPageMode: ShareMode;
   useIsSharedPage: (
     workspaceId: string,
     pageId: string
-  ) => [isSharePage: boolean, setIsSharePage: (enable: boolean) => void];
+  ) => {
+    isSharedPage: boolean;
+    toggleShare: (enable: boolean, shareMode?: ShareMode) => void;
+  };
+  useShareMode: (
```
Comment (joooye34) [inline packages/frontend/component/src/components/share-menu/share-menu.tsx:36]: Pass hook as props? so weird, maybe add comments to improve those code?

File: packages/frontend/core/src/components/page-detail-editor.tsx [hunk 3]
```diff
@@ -46,6 +46,9 @@ function useRouterHash() {
 
 export type OnLoadEditor = (page: Page, editor: EditorContainer) => () => void;
 
+function useRouterQuery() {
```
Comment (joooye34) [inline packages/frontend/core/src/components/page-detail-editor.tsx:49]: useSearchParams from `react-router-dom` ?

File: packages/frontend/component/src/components/share-menu/use-share-url.ts [hunk 4]
```diff
@@ -5,29 +5,40 @@ import { toast } from '../../ui/toast';
...
   pageId: string;
   urlType: UrlType;
+  mode?: ShareMode;
 };
 
 export const generateUrl = ({
   workspaceId,
   pageId,
   urlType,
+  mode = 'page',
```
Comment (joooye34) [inline packages/frontend/component/src/components/share-menu/use-share-url.ts:21]: if the mode has been saved in the DB, the value in the query seems unnecessary.

File: packages/frontend/core/src/hooks/affine/use-share-mode.ts [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+  getWorkspacePublicPagesQuery,
+  PublicPageMode,
+  publishPageMutation,
+} from '@affine/graphql';
+import { useAFFiNEI18N } from '@affine/i18n/hooks';
+import { useMutation, useQuery } from '@affine/workspace/affine/gql';
+import { useSetAtom } from 'jotai';
+import { useCallback, useMemo } from 'react';
+
+export function useShareMode(
```
Comment (joooye34) [inline packages/frontend/core/src/hooks/affine/use-share-mode.ts:13]: this hook seems so similar to `use-is-shared-page`, can we keep one?

File: packages/frontend/component/src/components/share-menu/use-share-url.ts (hunk 4 above)
Comment (joooye34) [reply packages/frontend/component/src/components/share-menu/use-share-url.ts:21]: the feature could redesign with mode data in the DB

File: packages/frontend/core/src/hooks/affine/use-is-shared-page.ts [hunk 6]
```diff
@@ -1,57 +1,138 @@
...
-            pageId,
+
+  const isSharedPage = useMemo(
+    () =>
+      data?.workspace.publicPages.some(publicPage => publicPage.id === pageId),
+    [data?.workspace.publicPages, pageId]
+  );
+
+  const toggleShare = useCallback(
+    (enable: boolean, mode?: ShareMode) => {
```
Comment (joooye34) [inline packages/frontend/core/src/hooks/affine/use-is-shared-page.ts:44]: split this function to changeShareMode and disableShare, making codes clearer.

File: packages/frontend/component/src/components/share-menu/use-share-url.ts (hunk 4 above)
Comment (JimmFly) [reply packages/frontend/component/src/components/share-menu/use-share-url.ts:21]: Do you mean removing the query when generating a link? Or does the link still carry the query, but the mode is returned by the database, hence no longer comes from an external input?

File: packages/frontend/component/src/components/share-menu/share-menu.tsx (hunk 2 above)
Comment (JimmFly) [reply packages/frontend/component/src/components/share-menu/share-menu.tsx]: I tried splitting it but that doesn't work, if you are not logged in the backend rejects the request and crashes the page.

**Comment by forehalo** [comment]
LGTM

File: packages/backend/server/src/modules/workspaces/controller.ts [hunk 7]
```diff
@@ -82,6 +84,20 @@ export class WorkspacesController {
       throw new NotFoundException('Doc not found');
     }
 
+    // fetch the publish page mode for publish page
```
Comment (forehalo) [inline packages/backend/server/src/modules/workspaces/controller.ts:87]: `if (!docId.isWorkspace)`

File: packages/frontend/graphql/src/graphql/public-page.gql [hunk 8]
```diff
@@ -0,0 +1,10 @@
+mutation publishPage(
```
Comment (forehalo) [inline packages/frontend/graphql/src/graphql/public-page.gql:1]: forget to delete the old public mutation schema

**Pull request merged**